DOCS-10107-db-execute-script-example-kt (#1278)
mulesoft/docs-connectors: db/1.9/modules/ROOT/pages/database-execute-script.adoc

= Execute a Script Examples - Mule 4

Anypoint Connector for Database (Database Connector) Execute Script operation executes a SQL script of random length as a single database statement.

> The Execute Script operation frequently creates schemas and tables, inserts data, and performs data rotation. Data rotation is performed by nightly jobs that move and archive data into historic tables and purge on-line tables. Some points to consider:

Reviewer: on-line > online, GLOBAL

* The script can contain multiple statements
* Statements can be of different types
* No input or output parameters are accepted

> * The operation runs any script that does not involve a SQL projection

Reviewer: End with periods because these are all sentences.

> You can use the Execute Script operation embedding the script in the operation, or reference the script from a file, but you cannot use both ways of executing a script simultaneously. +

Reviewer: ```suggestion
You can use the Execute Script operation either by embedding the script in the operation or by referencing the script from a file, but you cannot use both ways of executing a script simultaneously. +
```

> Executing a script returns an array of integer numbers, one element per each executed statement. Each number represents the number of objects affected by the statement.

Reviewer: ```suggestion
Executing a script returns an array of integers, one element per executed statement. Each integer represents the number of objects affected by the statement.
```

> The Execute Script and bulk operations are intended for different uses. For example, although you can build a script that inserts many rows into the database, you could not conveniently provide dynamic parameters to the script. Also, ensure that the script protects against a SQL injection attack.

Reviewer: How does the "Also," statement follow on either of the previous two?

> The following examples illustrates how to configure the Execute Script operation embedding the script in the operation, and how to reference the script from a file.

Reviewer: GLOBAL: review grammar rules around subject-verb agreement and fix.
Replace "examples illustrates" with
example illustrates
OR
examples illustrate

== Configure the Execute Script Operation Embedding the Script

To configure the operation in Studio, follow these steps:

. In the *Mule Palette* view, search for `database` and select the *Execute script* operation.
. Drag the *Execute script* operation onto the Studio canvas.
. In the *General* tab of the operation configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. Specify the database connection information and click *OK*.

> . Set the *SQL Query Text* field to the SQL query script to execute, for example:

Reviewer: ```suggestion
. Set the *SQL Query Text* field to the SQL query script to execute:
```

update PLANET set NAME='Mercury' where POSITION=4`.
----

The following screenshot shows the *Execute script* operation configuration:

.Execute script operation configuration
image::database-execute-script-1.png[Execute script operation configuration in Studio, in the SQL Query Text field set the SQL query script to execute]

In the XML editor, the `<db:execute-script>` configuration looks like this:

[source,xml,linenums]
----
<db:execute-script config-ref="dbConfig">
   <db:sql>
       update PLANET set NAME='Mercury' where POSITION=0;
       update PLANET set NAME='Mercury' where POSITION=4
   </db:sql>
</db:execute-script>
----

== Configure the Execute Script Operation Referencing the Script

To configure the Execute Script operation in Studio, follow these steps:

. In the *Mule Palette* view, search for `database` and select the *Select* operation.
. Drag the *Select* operation onto the Studio canvas.
. In the *General* tab of the operation configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. Specify the database connection information and click *OK*.

> . Set the *Script Path* field to the location of the script file to load, for example, `integration/executescript/bulk-script.sql`. +

Reviewer: insert blank line; the + must be on its own line, alone:
```suggestion
. Set the *Script Path* field to the location of the script file to load, for example, `integration/executescript/bulk-script.sql`. +
+
```
